### PR TITLE
Create user manager service

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Extract user handling from community manager into a user manager service

### CODE REVIEW

**Comment by wachterjohannes** [comment]
@alexander-schranz introduce interfaces for UserManager and CommunityManager.

File: Manager/UserManager.php [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+        // Create and Add User Role
+        $userRole = $this->createUserRole($user, $webspaceKey, $roleName);
+        $user->addUserRole($userRole);
+
+        // Save Entity
+        $this->entityManager->persist($userRole);
+        $this->entityManager->persist($contact);
+        $this->entityManager->persist($user);
+        $this->entityManager->flush();
```
Comment (wachterjohannes) [inline Manager/UserManager.php]: flush should be called in the controller to enable batch processing.

File: Manager/UserManager.php [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+    /**
+     * Find a user by a specific token.
+     *
+     * @param string $field
+     * @param string $token
+     *
+     * @return User
+     */
+    public function findByToken($field, $token)
```
Comment (wachterjohannes) [inline Manager/UserManager.php]: why did you pass the field here?

File: Manager/UserManager.php [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    {
+        return $this->userRepository->findOneBy([$field, $token]);
```
Comment (wachterjohannes) [inline Manager/UserManager.php]: this should be `['token' => $token]`

File: Manager/UserManager.php (hunk 2 above)
Comment (alexander-schranz) [inline Manager/UserManager.php]: could be `passwordResetToken` or `confirmationKey`

Comment (wachterjohannes) [inline Manager/UserManager.php]: why not adding two methods?

**Pull request merged**